### PR TITLE
Implement unitless length quirk

### PR DESCRIPTION
**Pull request by nox**
The Gecko side doesn't propagate its quirks mode yet.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16609)

### CODE REVIEW

File: components/style/values/specified/length.rs [hunk 1]
```diff
@@ -1154,20 +1154,20 @@ impl LengthOrPercentage {
...
                 NoCalcLength::parse_dimension(context, value.value, unit).map(LengthOrPercentage::Length),
             Token::Percentage(ref value) if num_context.is_ok(value.unit_value) =>
                 Ok(LengthOrPercentage::Percentage(Percentage(value.unit_value))),
-            Token::Number(ref value) => {
-                if value.value != 0. && !context.length_parsing_mode.allows_unitless_lengths() {
-                    return Err(())
-                }
-                Ok(LengthOrPercentage::Length(NoCalcLength::Absolute(AbsoluteLength::Px(value.value))))
-            }
+            Token::Number(ref value) if value.value == 0. ||
```
Comment (Manishearth) [inline components/style/values/specified/length.rs]: nit: parehtneses over precedence

**Comment by Manishearth** [comment]
r=me with those parens set.

**Comment by Wafflespeanut** [comment]
Haste Haste Haste... Test expectations need to be updated.

**Comment by nox** [comment]
Well this is good news, turns out all the tests pass on Linux, and the failing tests on macOS are unrelated.

@bors-servo r=Manishearth

**Comment by nox** [comment]
@bors-servo r=Manishearth

Tidy issue when rebasing.

**Comment by nox** [comment]
@bors-servo r=Manishearth

Build machine issues.

**Comment by nox** [comment]
I'm not sure whether `/html/semantics/interactive-elements/the-dialog-element/centering.html` is intermittent or not. I hope it is.

@bors-servo try

**Comment by nox** [comment]
Build is broken?

@bors-servo retry

**Comment by nox** [comment]
Build is broken?

@bors-servo retry

**Comment by chenpighead** [comment]
I believe you're hitting #16632... which I also hit it for hours.... :-(

**Pull request merged**